File: server/Models/User/Patient.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

[Table("patients")]
public class Patient
{
    [Key]
    public Guid Id { get; set; } = Guid.NewGuid();

    [Required]
    [MaxLength(30)]
    public string FirstName { get; set; } = string.Empty;

    [Required]
    [MaxLength(150)]
    public string LastName { get; set; } = string.Empty;

    [Required]
    [EmailAddress]
    public string Email { get; set; } = string.Empty;

    public DateTime? DateOfBirth { get; set; } = null;

    [MaxLength(150)]
    public string? Nationality { get; set; } = null;

    public char? Gender { get; set; } = null;

    [MaxLength(150)]
    public string? PlaceOfResidence { get; set; } = null;


    public int Is_Insurance { get; set; } = 0; //0: Dont have insurance, 1:Have Insurance

    public string? Address { get; set; } = null;

    [Required]
    [MaxLength(10)]
    public string PhoneNumber { get; set; } = string.Empty;

    public DateTime RegistrationDate { get; set; } = DateTime.Now;

    public UserAccount UserAccount { get; set; } = null!;
}

File: client/src/schemas/patient.ts
import { z } from "zod";

export const patientUpdateSchema = z.object({
  firstName: z.string().min(1, "Tên không được để trống"),
  lastName: z.string().min(1, "Họ và tên đệm không được để trống"),
  email: z
    .string()
    .min(1, "Email không được để trống")
    .email("Email không hợp lệ"),
  phoneNumber: z.string().min(10, "Số điện thoại phải có ít nhất 10 chữ số"),
  dateOfBirth: z.string().min(1, "Ngày sinh không được để trống"),
  nationality: z.string().min(1, "Quốc tịch không được để trống"),
  gender: z.string().min(1, "Giới tính không được để trống"),
  placeOfResidence: z.string().min(1, "Nơi sinh không được để trống"),
  address: z.string().min(1, "Địa chỉ thường trú không được để trống"),
  avatarUrl: z.string(),
});

export type PatientUpdateDto = z.infer<typeof patientUpdateSchema>;
